lib/SearchAndSort/SearchAndSort.js

```diff
@@ -152,6 +175,12 @@ class SearchAndSort extends React.Component {
       const resultAmount = nextProps.parentResources.records.other.totalRecords;
       this.SRStatus.sendMessage(`Search returned ${resultAmount} result${resultAmount !== 1 ? 's' : ''}`);
     }
+
+    if (this.state.filters.uninitialized && this.props.parentResources.query) {
+      this.setState({
+        filters: initialFilterState(this.props.filterConfig, this.props.parentResources.query.filters),
+      });
+    }
```

Comment:
This seems kind of perverted. What is the reasoning?

Reply:
To be DRY. I am not the author of initialFilterState, and at this point in the code we want to reset the filters to their initial state, whatever that was. This ensures that that happens in the same way it happened initially.

Reply:
Or at least I don't think I wrote initialFilterState...

Reply:
It could be that since the fix to stripes connect that this race condition goes away.

Reply:
Can you please try it? It would be a shame to commit this workaround if we don't need it.